### PR TITLE
Tim Sort Method in Java

### LINKED ISSUE #2158: Tim Sort
- [x] C
- [x] C++
- [ ] C#
- [ ] CoffeeScript
- [ ] Dart
- [ ] Go
- [ ] Java
- [x] JS
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by yuktirapartiwar**
Fixes #2158 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Tim Sort method in java

#### Languages Used:
- java

#### Files Added:
- Tim_Sort.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Tim_Sort/Tim_Sort.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
+// Tim sort implementation in java
+import java.util.Scanner;
+
```
Comment (asha15) [inline Tim_Sort/Tim_Sort.java]: remove  this line

File: Tim_Sort/Tim_Sort.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            k++;
+            i++;
+        }
+        while (j < rightLength) {
+            array[k] = Right[j];
+            k++;
+            j++;
+        }
+
+    }
```
Comment (asha15) [inline Tim_Sort/Tim_Sort.java]: keep a line space

File: Tim_Sort/Tim_Sort.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        n = scanner.nextInt();
+        int[] array = new int[n];
+        System.out.println("Enter your array: ");
+        for (int i = 0; i < n; i++) {
+            array[i] = scanner.nextInt();
+        }
+        System.out.println("Unsorted Array :");
+        for (int i = 0; i < n; i++)
+            System.out.println(array[i]);
+
```
Comment (asha15) [inline Tim_Sort/Tim_Sort.java]: Remove this line

File: Tim_Sort/Tim_Sort.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                merge(array, left, mid, right);
+            }
+        }
+
+        System.out.println("Sorted Array :");
+
+        for (int i = 0; i < n; i++)
+            System.out.println(array[i]);
+    }
+}
```
Comment (asha15) [inline Tim_Sort/Tim_Sort.java]: keep a line space after this

File: Tim_Sort/Tim_Sort.java [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+//        46
+//        58
+//        14
+//        Sorted Array :
+//        5
+//        14
+//        23
+//        46
+//        58
+//        78
```
Comment (asha15) [inline Tim_Sort/Tim_Sort.java]: make this as multi line commend

**Comment by yuktirapartiwar** [comment]
> Do this small changes, then we are good to merge

All the changes have been completed.

File: Tim_Sort/Tim_Sort.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        78
+        46
+        58
+        14
+        Unsorted Array :
+        23
+        5
+        78
+        46
+        58
```
Comment (asha15) [inline Tim_Sort/Tim_Sort.java]: no need to keep indentation in the output.  please remove the indentation in the commit

File: Tim_Sort/Tim_Sort.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        int n;
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Enter number of elements in your array: ");
+        n = scanner.nextInt();
+        int[] array = new int[n];
+        System.out.println("Enter your array: ");
+        for (int i = 0; i < n; i++) {
+            array[i] = scanner.nextInt();
+        }
+        System.out.println("Unsorted Array :");
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:67]: No need for printing unsorted array.

File: Tim_Sort/Tim_Sort.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
+// Tim sort implementation in java
+import java.util.Scanner;
+import static java.lang.Integer.min;
+
+public class Main {
+
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:6]: Remove this line

File: Tim_Sort/Tim_Sort.java [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+            array[k] = Left[i];
+            k++;
+            i++;
+        }
+        while (j < rightLength) {
+            array[k] = Right[j];
+            k++;
+            j++;
+        }
+
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:40]: Remove this line

File: Tim_Sort/Tim_Sort.java [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+public class Main {
+
+    // Merge Sort function
+    public static void merge(int[] array, int left, int mid, int right) {
+        int leftLength = mid - left + 1;
+        int rightLength = right - mid;
+        int[] Left = new int[leftLength];
+        int[] Right = new int[rightLength];
+        for (int i = 0; i < leftLength; i++)
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:13]: leave a line before it

File: Tim_Sort/Tim_Sort.java [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+            Left[i] = array[left + i];
+        for (int i = 0; i < rightLength; i++)
+            Right[i] = array[mid + 1 + i];
+        int i = 0;
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:17]: leave a line before it

File: Tim_Sort/Tim_Sort.java [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+        int j = 0;
+        int k = left;
+        while (i < leftLength && j < rightLength) {
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:20]: leave a line before it

File: Tim_Sort/Tim_Sort.java [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+            if (Left[i] <= Right[j]) {
+                array[k] = Left[i];
+                i++;
+            } else {
+                array[k] = Right[j];
+                j++;
+            }
+            k++;
+        }
+        while (i < leftLength) {
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:30]: leave a line before it

File: Tim_Sort/Tim_Sort.java [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+            array[k] = Left[i];
+            k++;
+            i++;
+        }
+        while (j < rightLength) {
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:35]: leave a line before it

File: Tim_Sort/Tim_Sort.java [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+            while (j >= left && array[j] > temp) {
+                array[j + 1] = array[j];
+                j--;
+            }
+            array[j + 1] = temp;
+        }
+    }
+
+    public static void main(String[] args) {
+
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:57]: Remove this line

File: Tim_Sort/Tim_Sort.java [hunk 16]
```diff
@@ -0,0 +1,111 @@
...
+        int n;
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Enter number of elements in your array: ");
+        n = scanner.nextInt();
+        int[] array = new int[n];
+        System.out.println("Enter your array: ");
```
Comment (somya-kapoor) [inline Tim_Sort/Tim_Sort.java:63]: leave a line before it

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by yuktirapartiwar** [comment]
> Are you working on it?

changes resolved

**Comment by somya-kapoor** [comment]
Please squash the commits

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this PR

**Pull request closed without merging**